build.cake

```diff
@@ -578,7 +643,7 @@ void BuildProject(string projectPath, string configuration, MSBuildPlatform buil
     }
 }
  
-void RunTest(FilePath exePath, DirectoryPath workingDir)
+void RunTest(FilePath exePath, DirectoryPath workingDir, string framework, ref List<string> errorDetail)
```

Comment:
This is to do with mono/linux compatibility, in .NET/windows from within the method you can access the variables,  but in mono you cannot. So i need to pass this by ref in to get modified. Objects default to getting passed by val (https://msdn.microsoft.com/en-us/library/14akc2c7.aspx), so using ref the modifications to the object will be carried out of the method in the event of an error. The list compiles a collection of all errors to be outputted at the end of the user.